### PR TITLE
Add warning flash message

### PR DESCRIPTION
**Pull request by kfarst**
Needed for https://www.pivotaltracker.com/story/show/102410476

screen shot 2015-08-31 at 10 36 23 pm

### CODE REVIEW

File: src/angular-gs-flash-message.js [hunk 1]
```diff
@@ -23,13 +23,19 @@ angular.module('gs.flash-message', ['ngSanitize'])
...
     template:
-      '<div class="alert" ng-if="messages.error.length || messages.success.length">' +
+      '<div class="alert" ng-if="messages.error.length || messages.success.length || messages.warning.length">' +
         '<div ng-if="messages.error.length" class="alert--message alert--message__error">' +
           '<span class="alert--message--content" ng-repeat="error in messages.error track by $index" ng-bind-html="error"></span>' +
           '<a href="javascript:;" ng-click="close()">' +
             '<i class="alert--message--remove-icon"></i>' +
           '</a>' +
         '</div>' +
+        '<div ng-if="messages.warning.length" class="alert--message alert--message__warning">' +
```
Comment (tjwallace) [inline src/angular-gs-flash-message.js]: I wonder if you could look through `messages` instead of explicitly checking `error`, `success`, `length`, etc?

**Comment by tjwallace** [comment]
Small nitpick, otherwise +1

File: src/angular-gs-flash-message.js (hunk 1 above)
Comment (kfarst) [inline src/angular-gs-flash-message.js]: Do you mean iterating over the object keys and interpolating those keys into the message fetching and CSS classes?

Comment (tjwallace) [inline src/angular-gs-flash-message.js]: Yes, exactly. If you want to ensure order you might have to look over an array and use the value to check the `messages` object.

File: src/angular-gs-flash-message.js [hunk 2]
```diff
@@ -23,13 +23,19 @@ angular.module('gs.flash-message', ['ngSanitize'])
...
+          '<span class="alert--message--content" ng-repeat="warning in messages.warning track by $index" ng-bind-html="warning"></span>' +
+          '<a href="javascript:;" ng-click="close()">' +
```
Comment (evanhourigan) [inline src/angular-gs-flash-message.js]: I think this can just be `<a href ng-click="close()">` (no value for the `href` attribute)

Comment (kfarst) [inline src/angular-gs-flash-message.js]: done in e29efd9

**Comment by kfarst** [comment]
@tjwallace @evanhourigan updated again and changed commit message to be more specific

**Comment by tjwallace** [comment]
+2

**Pull request merged**